core/core.go

```diff
@@ -293,6 +293,14 @@ func setupDiscoveryOption(d config.Discovery) DiscoveryOption {
 			}
 			return discovery.NewMdnsService(h, time.Duration(d.MDNS.Interval)*time.Second)
 		}
+	} else if d.Cjdns.Enabled {
+		return func(h p2phost.Host) (discovery.Service, error) {
+			interval := d.Cjdns.Interval
+			if interval == 0 {
+				interval = 5
+			}
+			return discovery.NewCjdnsService(h, time.Duration(interval)*time.Second)
+		}
```

Comment:
setupDiscoveryOption will need a refactoring. Right now it can only use either MDNS *or* Cjdns. This is also a bad place for the default interval.